stacks_queues/queue.py

```diff
@@ -15,47 +15,75 @@ def __init__(self):
         self.front = -1
         self.rear = -1
         self.size = 0
-      
+
 
     def enqueue(self, element):
         """ Adds an element to the Queue
             Raises a QueueFullException if all elements
             In the store are occupied
             returns None
         """
-        pass
+        if self.empty():
+            self.front = 0
+            self.rear = 0
+        elif self.rear == self.front:
+            raise QueueFullException("Queue is full!")
+        self.store[self.rear] = element
+        self.rear = (self.rear + 1) % self.buffer_size
 
     def dequeue(self):
         """ Removes and returns an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
         """
-        pass
+        if self.empty():
+            raise QueueEmptyException("Queue is empty!")
+        else:
+            result = self.store[self.front]
+        self.front = (self.front + 1) % self.buffer_size
+        if self.front == self.rear:
+            self.front = -1
```

Comment:
This indicates the queue is empty, so you might also want to reset `self.rear`
```suggestion
        if self.front == self.rear:
            self.front = -1
            self.rear = -1
```